test/e2e/core.go

```diff
@@ -97,3 +110,29 @@ func logCore(cmds []command, hosts []string, dir, provider string) {
 	}
 	wg.Wait()
 }
+
+func isUsingSystemdKubelet(provider string, hosts ...string) bool {
+	wg := &sync.WaitGroup{}
+	results := make([]int, len(hosts))
+	cmd := "sudo systemctl status kubelet.service"
+
+	wg.Add(len(hosts))
+	for i := range hosts {
+		go func(i int) {
+			defer wg.Done()
+			result, err := SSH(cmd, hosts[i], provider)
+			if err != nil {
+				fmt.Printf("Error running command: %v\n", err)
+			}
+			results[i] = result.Code
```

Comment:
@jonboulle raises a good point. if a node has gone away (and thus this ssh call fails), we don't want that to influence the choice of logging location for the rest of the nodes.

I think we should probably be using only successful ssh connections to detect whether nodes are using systemd or not. Could probably even shortcut - return true/false for the first host that we successfully connect to, rather than checking all. I imagine that's the easiest fix.